docs/how-tos/create-nexus-files-by-python-and-validate-them.md

```diff
@@ -0,0 +1,940 @@
+ERROR    1     violates NeXus specification                                -10000000
+TODO     16    validation not implemented yet                              0        
+UNUSED   0     optional NeXus item not used in data file                   0        
+COMMENT  0     comment from the punx source code                           0        
+OPTIONAL 213   allowed by NeXus specification, not identified              99       
+         --                                                                         
+TOTAL    378                                                                        
+======== ===== =========================================================== =========
+```
+
+The last error message:
+
+```
+======= ====== ========== ======================================
+/entry  ERROR  known NXDL NXopt: unrecognized NXDL specification
+======= ====== ========== ======================================
+```
+
+can be ignored and is a bug right now. If this is the only Error message, then your NeXus file is compliant with the NeXus definitions and you can share and publish your data.
+
+# 3\.3.4. Further Notes
+
+1. Punx only uses the NeXus definiton from the NIAC ([https://manual.nexusformat.org/](<https://manual.nexusformat.org/>)) - The FAIRmat NeXus definition is not possible right now ([https://fairmat-nfdi.github.io/nexus\_definitions/index.html#](<https://fairmat-nfdi.github.io/nexus_definitions/index.html#>))
+
+2. Other punx commands are avaialble: [https://punx.readthedocs.io/en/latest/overview.html#](<https://punx.readthedocs.io/en/latest/overview.html#>)
+
+3. More details for installation: [https://punx.readthedocs.io/en/latest/install.html](<https://punx.readthedocs.io/en/latest/install.html>)
+
+4. Github project: [https://github.com/prjemian/punx](<https://github.com/prjemian/punx>)
+
+
+
+
+
+# 3\.4 pynxtools - Python Nexus Tools
+
+This is python package which is developed by the FAIRmat consortium.
+
+As python package, this can be used on Linux and Windows systems.
+
+The package can be installed via pip. Therefore you need to have installed:
+
+1. python
+
+2. pip
+
+This tool has 3 command line functions:
+
+1. dataconverter ([https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/dataconverter/README.md](<https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/dataconverter/README.md>))
+
+2. read\_nexus ([https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/nexus/README.md](<https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/nexus/README.md>))
+
+3. generate\_eln ([https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/eln\_mapper/README.md](<https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/eln_mapper/README.md>))
+
+For validation purposes, we will use the "read\_nexus" function.
+
+The command used is:
+
+```
+read_nexus -f NXopt_minimal_example.nxs
```

Comment:
eventually, this shall be replaced by the `verify_nexus` command from #333

Reply:
I added the respective section and kept for now as well read_nexus, as verify_nexus is still in development.

Reply:
We should keep both eventually, they do indeed different things